engine/src/main/java/org/terasology/engine/math/Side.java

```diff
@@ -116,10 +116,8 @@ public static Side inHorizontalDirection(double x, double z) {
      * method.
      *
      * @return All available sides'
-     * @deprecated use {@link Side#values()}
      */
-    @Deprecated
-    public static EnumSet<Side> getAllSides() {
+    public static List<Side> getAllSides() {
```

Comment:
looks like values() might add a lot of overhead the only way to make this "immutable" is to always return a new array :/. will have to use ImmutableList.

Reply:
@pollend Does this mean, you'll do another commit on this?

Reply:
will have to rework what I considered deprecated because it adds a lot of overhead. :/

Reply:
[Memory-Hogging Enum.values() Method](https://dzone.com/articles/memory-hogging-enumvalues-method) is a good read on that topic. 

After reading through this, I think we should keep the cached collection of values and add docstrings to explain why we're doing this, and why it is preferable to use `getAllSides()` over `.values()`. 

@pollend why did you change the [EnumSet](https://docs.oracle.com/javase/7/docs/api/java/util/EnumSet.html) to a `List`? Is it to guarantee immutability? If we're concerned about the footprint of the collection, we could also use [`Guava#immutableEnumSet`](https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/Sets.html#immutableEnumSet(E,E...))

Reply:
@pollend could you answer this, please?

Reply:
an enumset tries to pack the data wonder if that has an impact on performance.